node_modules/oae-authentication/lib/strategies/google/init.js

```diff
@@ -56,16 +69,28 @@ module.exports = function() {
             log().trace({
                 'tenant': tenant,
                 'profile': profile
-            }, 'Received Google authentication callback.');
+            }, 'Received Google authentication callback');
 
             var email = profile.emails[0].value;
 
-            // Re-use the email address as the externalId.
+            // Ensure the email belongs to a domain we allow
+            if (!_.isEmpty(domains)) {
+                var emailDomain = email.split('@')[1];
+                if (!_.contains(domains, emailDomain)) {
+                    var err = {
+                        'message': util.format('You tried to sign in with an that belongs to a domain (%s) that is not allowed access', emailDomain),
+                        'reason': 'domain_not_allowed'
```

Comment:
It's hard to pass information from an OAE strategy through passport back to OAE. This reason key will get checked by AuthenticationUtil.handlePassportError. Because there's an error, authentication will fail and the user will be redirected to: `/?authentication=failed&reason=domain_not_allowed`.

If we ever get support tickets or we want to show an appropriate error message, that reason value could be used.